model/pod/pod.go

```diff
@@ -156,20 +184,100 @@ func (p *Pod) Remove() error {
 	)
 }
 
-// UpdateStatus updates the pod status.
+// UpdateStatus updates the pod status. If the new status is identical to the
+// current one, this is a no-op.
 func (p *Pod) UpdateStatus(s Status) error {
+	if p.Status == s {
+		return nil
+	}
+
 	byIDAndStatus := ByID(p.ID)
 	byIDAndStatus[StatusKey] = p.Status
 
+	updated := utility.BSONTime(time.Now())
+	setFields := bson.M{StatusKey: s}
+	switch s {
+	case StatusInitializing:
+		setFields[bsonutil.GetDottedKeyName(TimeInfoKey, TimeInfoInitializingKey)] = updated
+	case StatusStarting:
+		setFields[bsonutil.GetDottedKeyName(TimeInfoKey, TimeInfoStartingKey)] = updated
+	}
+
 	if err := UpdateOne(byIDAndStatus, bson.M{
-		"$set": bson.M{StatusKey: s},
+		"$set": setFields,
 	}); err != nil {
 		return err
 	}
 
 	event.LogPodStatusChanged(p.ID, string(p.Status), string(s))
 
 	p.Status = s
+	switch s {
+	case StatusInitializing:
+		p.TimeInfo.Initializing = updated
+	case StatusStarting:
+		p.TimeInfo.Starting = updated
+	}
 
 	return nil
 }
+
+// UpdateResources updates the pod resources. If the new status is identical to the current one, this is a no-op.
+func (p *Pod) UpdateResources(info ResourceInfo) error {
+	if p.Resources.Cluster == info.Cluster && p.Resources.ExternalID == info.ExternalID && p.Resources.DefinitionID == info.DefinitionID && p.checkSecretIDs(p.Resources.SecretIDs, info.SecretIDs) {
```

Comment:
I don't think we actually have to check this. `UpdateStatus` only checks this as a small optimization but it's not typical to check that we're doing a no-op update.